samples/client/petstore/go/go-petstore/api_fake.go

```diff
@@ -834,11 +834,7 @@ func (a *FakeApiService) TestEndpointParameters(ctx _context.Context, number flo
 		localVarFormParams.Add("date", parameterToString(localVarOptionals.Date.Value(), ""))
 	}
 	if localVarOptionals != nil && localVarOptionals.DateTime.IsSet() {
-		paramJson, err := parameterToJson(localVarOptionals.DateTime.Value())
-		if err != nil {
-			return nil, err
-		}
-		localVarFormParams.Add("dateTime", paramJson)
+		localVarFormParams.Add("dateTime", parameterToString(localVarOptionals.DateTime.Value(), ""))
```

Comment:
The fix is good:
```
        - name: date
          type: string
          format: date
          in: formData
          description: None
        - name: dateTime
          type: string
          format: date-time
          in: formData
          description: None
```
so there shouldn't be any nil check as the parameter is not required.